tda/auth.py

```diff
@@ -163,10 +164,79 @@ def client_from_login_flow(webdriver, api_key, redirect_url, token_path,
 
     if asyncio:
         session_class = AsyncOAuth2Client
-        client_class  = AsyncClient
+        client_class = AsyncClient
+    else:
+        session_class = OAuth2Client
+        client_class = Client
+
+    # Return a new session configured to refresh credentials
+    return client_class(
+        api_key,
+        session_class(api_key, token=token,
+                      auto_refresh_url='https://api.tdameritrade.com/v1/oauth2/token',
+                      auto_refresh_kwargs={'client_id': api_key},
+                      update_token=update_token))
+
+
+def client_from_login_flow_terminal(api_key, redirect_url, token_path, asyncio=False):
+    '''
+    Uses the terminal to perform an OAuth webapp login flow and creates a
+    client wrapped around the resulting token. The client will be configured to
+    refresh the token as necessary, writing each updated version to
+    ``token_path``.
+
+    :param api_key: Your TD Ameritrade application's API key, also known as the
+                    client ID.
+    :param redirect_url: Your TD Ameritrade application's redirect URL. Note
+                         this must *exactly* match the value you've entered in
+                         your application configuration, otherwise login will
+                         fail with a security error.
+    :param token_path: Path to which the new token will be written. If the token
+                       file already exists, it will be overwritten with a new
+                       one. Updated tokens will be written to this path as well.
+    '''
+    get_logger().info(('Creating new token with redirect URL \'{}\' ' +
+                       'and token path \'{}\'').format(redirect_url, token_path))
+
+    api_key = __normalize_api_key(api_key)
+
+    oauth = OAuth2Client(api_key, redirect_uri=redirect_url)
+    authorization_url, state = oauth.create_authorization_url(
+        'https://auth.tdameritrade.com/auth')
+
+    # Open the login page and wait for the redirect
+    print('\n**************************************************************\n')
+    print('Open the following page in a browser. Login, and then paste the URL that your browser is sent to below.')
+    print()
+    print(authorization_url)
+    print()
+    print('If you encounter any issues, see here for troubleshooting: ' +
+          'https://tda-api.readthedocs.io/en/stable/auth.html' +
+          '#troubleshooting')
+    print('\n**************************************************************\n')
+
+    current_url = input("Redirected URL:  ")
```

Comment:
I tested your function on Mac and it seems not to work. For some reason python's input() hangs for large inputs. Instead trying the prompt function from `prompt_toolkit` works flawlessly.

Reply:
That's quite weird. For me, on MacOS as well, I experienced the same hang on Python 3.8, but was fixed after importing the "readline" module. Let me test again with 3.9 on MacOS and figure out if we need to pull in the external module.